### PR TITLE
Fixed various incorrect usages of defined macros

### PR DESCRIPTION
**Pull request by Zarickan**
For example, in fn_initServer.sqf Info_2 is defined with 3 parameters, but in this case it is passed (unintentially) 4 arguments, this means that the macro will instead be expanded to nothing.

Related documentation can be found on the [wiki](https://community.bistudio.com/wiki/PreProcessor_Commands), relevant part is under __Arguments__.

## What type of PR is this.
1. [x] Bug
2. [ ] Change
3. [ ] Enhancement

### Please verify the following and ensure all checks are completed.
1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Comment by Bob-Murphy** [comment]
Hi Zarickan,
thank you for the PR and sorry for the delay.
Somebody of our team will look at this shortly.
Cheers

File: A3-Antistasi/functions/AI/fn_artySupport.sqf [hunk 1]
```diff
@@ -257,7 +257,8 @@ if (_typeArty != "BARRAGE") then
 	_eta = (_artyArrayDef1 select 0) getArtilleryETA [_positionTel, ((getArtilleryAmmo [(_artyArrayDef1 select 0)]) select 0)];
 	_timeX = time + _eta - 5;
 	if (isNil "_timeX") exitWith {
-        Error_4("Params: %1,%2,%3,%4",_artyArrayDef1 select 0,_positionTel,((getArtilleryAmmo [(_artyArrayDef1 select 0)]) select 0),(_artyArrayDef1 select 0) getArtilleryETA [_positionTel, ((getArtilleryAmmo [(_artyArrayDef1 select 0)]) select 0)]);
+#define ARTILLERY_ERROR_INFORMATION [_positionTel, ((getArtilleryAmmo [(_artyArrayDef1 select 0)]) select 0)]
```
Comment (killerswin2) [inline A3-Antistasi/functions/AI/fn_artySupport.sqf:260]: I'm okay with this being a define, but move it to an hpp file. lets keep the sqf files free from macro defines.

File: A3-Antistasi/functions/Missions/fn_convoy.sqf [hunk 2]
```diff
@@ -283,7 +283,8 @@ private _fnc_applyResults =
         killZones setVariable [_mrkOrigin,_killZones,true];
     };
 
-    ServerDebug_2("Rebels %1 a %2 convoy mission", ["lost", "won"] select _success, _type);
+#define CONVOY_MISSION_STATUS_ARRAY ["lost", "won"]
```
Comment (killerswin2) [inline A3-Antistasi/functions/Missions/fn_convoy.sqf:286]: Again move to a hpp

File: A3-Antistasi/functions/init/fn_initServer.sqf [hunk 3]
```diff
@@ -84,7 +84,8 @@ _nul = call A3A_fnc_initVar;
 call A3A_fnc_logistics_initNodes;
 
 savingServer = true;
-Info_2("%1 server version: %2", ["SP","MP"] select isMultiplayer, localize "STR_antistasi_credits_generic_version_text");
+#define MODE_ARRAY ["SP","MP"]
```
Comment (killerswin2) [inline A3-Antistasi/functions/init/fn_initServer.sqf:87]: hpp file, please

**Review by killerswin2 — CHANGES_REQUESTED** [review]
All good changes, just keep the sqf files clear from preprocessor macro defines

**Comment by CalebSerafin** [comment]
For those array cases, we could pull a page out of CBA and ACE.

in `A3-Antistasi\Includes` we could create a file called `generalMacro.hpp` or something else that sounds appropriate.

We include it in common.hpp

Then we can pillage https://github.com/CBATeam/CBA_A3/blob/876b9b2f6afa7f609d58955333078531a33f9836/addons/main/script_macros_common.hpp#L1066-L1084

After that, we can use to to inline arrays like so:
```cpp
INFO_1("Test : %1", [ARR("false","true")] select true)
```

**Comment by HakonRydland** [comment]
Those macros will be a part of move to mod, in `script_macros_common.hpp`

**Comment by CalebSerafin** [comment]
Move to mod is not happing yet, so this will be a suitable solution for now :)

**Review by CalebSerafin — APPROVED** [review]
After moving to mod we can weed out these special cases that needed their own array macro.
For now, it isn't too much of a war crime leaving it as is.

**Review by killerswin2 — APPROVED** [review]
we will change to a hpp once mod to move it complete

**Pull request merged**